python/ray/_raylet.pyx

```diff
@@ -1035,23 +1036,12 @@ cpdef report_streaming_generator_output(
     worker = ray._private.worker.global_worker
 
     cdef:
-        CoreWorker core_worker = worker.core_worker
         # Ray Object created from an output.
         c_pair[CObjectID, shared_ptr[CRayObject]] return_obj
 
-    try:
-        if isinstance(output_or_exception, Exception):
-            raise output_or_exception
-    except AsyncioActorExit:
-        # Make the task handle this exception.
-        raise
-    except StopAsyncIteration:
-        return True
-    except StopIteration:
-        return True
-    except Exception as e:
+    if isinstance(output_or_exception, Exception):
```

Comment:
if I raise an exception again like before, it caused the circular dependencies issues.